mealie/core/dependencies/dependencies.py

```diff
@@ -205,24 +204,15 @@ def validate_recipe_token(token: str | None = None) -> str:
     return slug
 
 
-async def temporary_zip_path() -> AsyncGenerator[Path, None]:
+def get_temporary_zip_path() -> Path:
     app_dirs.TEMP_DIR.mkdir(exist_ok=True, parents=True)
-    temp_path = app_dirs.TEMP_DIR.joinpath("my_zip_archive.zip")
-
-    try:
-        yield temp_path
-    finally:
-        temp_path.unlink(missing_ok=True)
+    return app_dirs.TEMP_DIR.joinpath("my_zip_archive.zip")
 
 
-async def temporary_dir(background_tasks: BackgroundTasks) -> AsyncGenerator[Path, None]:
+def get_temporary_dir() -> Path:
```

Comment:
Can we make this a context manager so we don't have to unlink the file every time this is called? Something like:

```python
from contextlib import contextmanager
...
@contextmanager
def get_temporary_path(auto_unlink=True) -> Path:  # also if we could rename this, since it's a temp path not a temp dir
    temp_path = app_dirs.TEMP_DIR.joinpath(uuid4().hex)
    temp_path.mkdir(exist_ok=True, parents=True)

    try:
        yield temp_path
    finally:
        if auto_unlink:
            temp_path.unlink(missing_ok=True)
```

Then in the code use it like this:
```python
with get_temporary_dir() as temp_dir:
    ...
```